### PR TITLE
Fix zero argument result allocation for nested power manifold

### PR DESCRIPTION
**Pull request by mateuszbaran**
A small fix needed for https://github.com/JuliaManifolds/Manifolds.jl/pull/467 .

### CODE REVIEW

File: src/PowerManifold.jl [hunk 1]
```diff
@@ -200,7 +200,7 @@ function Base.:^(
 end
 
 function allocate_result(M::PowerManifoldNested, f, x...)
-    if representation_size(M.manifold) === ()
```
Comment (kellertuer) [inline src/PowerManifold.jl:203]: can you explain what case this does change (in pratice/examples)?

File: src/PowerManifold.jl [hunk 2]
```diff
@@ -209,8 +209,12 @@ function allocate_result(M::PowerManifoldNested, f, x...)
         ]
     end
 end
-function allocate_result(::PowerManifoldNestedReplacing, f, x...)
-    return copy(x[1])
```
Comment (kellertuer) [inline src/PowerManifold.jl:213]: Same question as above – just curious :)

Comment (mateuszbaran) [reply src/PowerManifold.jl:213]: What was failing before is `allocate_result(PowerManifold(...), rand)`, because there was no `x[1]`.

Comment (kellertuer) [reply src/PowerManifold.jl:213]: Ah! sure – that makes sense.

**Review by kellertuer — APPROVED** [review]

**Pull request merged**